src/EnvRegistry.ts

```diff
@@ -54,10 +51,10 @@ export class EnvRegistry implements EnvRegistryInterface {
     return this.save({ key: envKey, value: env });
   }
 
-  public environments$(environmentsRequest: EnvironmentsRequest): EnvironmentsResponse {
+  public environments$(environmentsRequest: EnvironmentsRequest): EnvironmentsResponse<T> {
     return from(this.configurationService.entries({ repository: this.repository })).pipe(
       switchMap((response: EntriesResponse) => {
-        return from(response.entries.map((entry) => ({ envKey: entry.key, env: entry.value })));
+        return from(response.entries.map((entry) => ({ envKey: entry.key, env: entry.value } as EnvRegistryItem<T>)));
```

Comment:
It seems like a cheating here, when you force the value to be treated as a type.
ConfigurationService allows to pass type variable inside (any by default). I think it can help you to tell TS that env is with the type Env.

Reply:
Actually it was not there before, I don't remember why I added it. I will remote it it's not needed

Reply:
If possible, we have to avoid these constructions